tasks/main.yml

```diff
@@ -24,9 +24,21 @@
     ( ansible_distribution_major_version > "7" and
       ( ansible_distribution == "CentOS" or ansible_distribution == "RedHat" ))
 
-- name: Set permanent SELinux state
-  selinux: policy={{ selinux_policy }} state={{ selinux_state }}
-  when: selinux_state is defined
+- name: Set permanent SELinux state if enabled
+  selinux:
+    state: "{{ selinux_state | default(ansible_selinux.config_mode) }}"
+    policy: "{{ selinux_policy | default(ansible_selinux.type) }}"
+  when: ansible_selinux.status == "enabled"
+
+- name: Set permanent SELinux state is disabled
+  selinux:
+    state: "{{ selinux_state }}"
+    policy: "{{ selinux_policy | default('targeted') }}"
+  when: ansible_selinux.status == "disabled" and selinux_state is defined
+
+- debug:
+    msg: "SELinux is disabled on system and current playbook does not enable it, some SELinux modules can crash"
```

Comment:
I thing this message needs to be reformulated. The playbook could enable it using `selinux_state` in task before but it would take effect after reboot.

Reply:
Sorry, Don't understand what you mean. Could you add example how it should be reformulated?

Reply:
"SELinux is disabled on system - some SELinux modules can crash"  would be enough. The part "current playbook does not enable it" could be confusing when playbook sets `selinux_state` to `enforcing`

Reply:
Fixed.

Reply:
>  The playbook could enable it using selinux_state in task before but it would take effect after reboot.

In that case we will fail earlier and don't reach this message, do we?